internal/databases/mysql/binlog_push_handler.go

```diff
@@ -68,32 +69,45 @@ func HandleBinlogPush(uploader internal.UploaderProvider, untilBinlog string, ch
 		filter.init(cache)
 	}
 
-outer:
 	for i := 0; i < len(binlogs); i++ {
 		binLog := binlogs[i]
 
 		tracelog.DebugLogger.Printf("Testing... %v\n", binLog)
+		upload := true
 		for _, filter := range filters {
 			nextBinLog := ""
 			if i < len(binlogs)-1 {
 				nextBinLog = binlogs[i+1]
 			}
 			if !filter.test(binLog, nextBinLog) {
 				tracelog.DebugLogger.Printf("Skip binlog %v (%s check)\n", binLog, filter.name())
-				continue outer
+				upload = false
+				break
 			}
 		}
 
+		if !upload {
+			for _, filter := range filters {
+				filter.onSkip(&cache)
```

Comment:
We should not trigger onSkip and modify `LastArchivedBinlog` if binlog was skipped by  `archivedBinlogFilter`

Suppose we have `bl.001` - `bl.099`, and all binlogs up to `bl.050` were already uploaded and `LastArchivedBinlog` is set to `bl.050` in cache.
During next run we will check `bl.001` to be uploaded, `archivedBinlogFilter` will discard it, and we will update `data.LastArchivedBinlog`  back to `bl.001`.